pygmt/figure.py

```diff
@@ -374,6 +244,9 @@ def savefig(  # noqa: PLR0912
                 raise GMTInvalidInput(msg)
             kwargs["W"] = True
 
+        # pytest-mpl v0.17.0 added the "metadata" parameter to Figure.savefig, which
```

Comment:
These lines were initially added to `Figure.psconvert` in #3054, but I feel it makes more sense to add the lines in `Figure.savefig` instead.